packages/@o3r-training/tools/cli/extract-folder-structure/extract-folder-structure.ts

```diff
@@ -0,0 +1,29 @@
+import {lstat, readdir, readFile, writeFile} from 'node:fs/promises';
+import {type FileSystem, getFilesTree} from '@o3r-training/tools';
+import {join, resolve} from 'node:path';
+import {program} from 'commander';
+
+program
+  .description('Extract folder structure')
+  .requiredOption('--files <files>', 'List of files and folder to extract in addition to the path')
+  .option('-r, --root <root>', 'Root of the extraction')
+  .option('-o, --output <output>', 'Output folder path')
+  .parse(process.argv);
+
+const options: any = program.opts();
+const cwd = options.root ? resolve(process.cwd(), options.root) : process.cwd();
+void (async () => {
+  const filesDescriptor = [];
+  for (const file of options.files.split(',')) {
+    const filePath = join(cwd, file);
+    filesDescriptor.push({isDir: (await lstat(file)).isDirectory(), path: filePath});
+  }
```

Comment:
```suggestion
  const filesDescriptor = await Promise.all(
    options.files.split(',').map(async (file) => {
      const filePath = join(cwd, file);
      return {isDir: (await lstat(file)).isDirectory(), path: filePath};
    })
  );
```
(because there is no need to access to the files in the same order)